jsonapi/unmarshal.go

```diff
@@ -200,7 +221,13 @@ func setRelationshipIDs(relationships map[string]Relationship, target UnmarshalI
 			if !ok {
 				return fmt.Errorf("struct %s does not implement UnmarshalToOneRelations", reflect.TypeOf(target))
 			}
-			err := castedToOne.SetToOneReferenceID(name, rel.Data.DataObject.ID)
+
+			var err error
+			if rel.Data.DataObject.ID == "" {
+				err = castedToOne.SetToOneReferenceID(name, rel.Data.DataObject.LID)
```

Comment:
I guess that is not correct to set LID to reference ID. Let's create the another interface with function `SetToOneReferenceIDWithLID`